### PR TITLE
Bookmarks: Use feature availablity checks

### PR DESCRIPTION
**Pull request by ashiagr**
## Description

This cherry picks feature availability changes from #1332. 

## Testing Instructions

1. Make sure that tests are run and CI is green
2. Head over to PR #1419 to test the changes for the bookmarks feature.

## Merge Instructions

1. Make sure that PR #1419 is approved
2. Remove `not ready for merge` label
3. Merge the PR.

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [ ] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by ashiagr** [comment]
@mchowning, can you review this commit https://github.com/Automattic/pocket-casts-android/pull/1418/commits/06a9dacb9400c17bf9f77acf94ba78bd72b0fae4? 

Changes in https://github.com/Automattic/pocket-casts-android/pull/1418/commits/bb0158ca70cc6b639e225ca6672d0bdcef3fa5f0 look great and tests pass too. Since the changes are tested in #1419, the PR should be ready to merge once all commits are approved.

**Review by mchowning — APPROVED** [review]
Looks good and works well. Just a couple of small comments, but there aren't any required changes.

File: modules/services/featureflag/src/main/java/au/com/shiftyjelly/pocketcasts/featureflag/Feature.kt
Comment (ashiagr) [reply modules/services/featureflag/src/main/java/au/com/shiftyjelly/pocketcasts/featureflag/Feature.kt]: Enabled for debug build: https://github.com/Automattic/pocket-casts-android/pull/1418/commits/35c6b9afba740f32a39b5ef71ecdf4123da90adb

File: modules/services/featureflag/src/test/java/au/com/shiftyjelly/pocketcasts/featureflag/FeatureTest.kt
Comment (ashiagr) [reply modules/services/featureflag/src/test/java/au/com/shiftyjelly/pocketcasts/featureflag/FeatureTest.kt:84]: I can't think of a good way to address it right now. I've created a [task](https://github.com/Automattic/pocket-casts-android/issues/1428 ) to track it.

**Pull request merged**
